artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -93,8 +93,15 @@ impl TryConvertMut<&mut [Value], Args> for Artichoke {
                     // ```
                     let month: i64 = if let Ok(arg) = to_str(self, arg) {
                         let mut month_str: Vec<u8> = arg.try_convert_into_mut(self)?;
+
+                        // Valid month string args are always 3 bytes long (or 3
+                        // ASCII characters). only downcase the first 3 bytes to
+                        // avoid excessive resource consumption if given a long
+                        // string.
+                        let month_str = month_str.get_mut(..3).unwrap_or_default();
```

Comment:
I don't want to ruin your day, but:

```irb
3.1.2 > class I; def to_str; "0000000002"; end; end
=> :to_str
3.1.2 > Time.utc(2022, I.new)
=> 2022-02-01 00:00:00 **UTC**
```

:sweat:

Reply:
this should still work, right? because the slice only checks for the month names. If none of those match, the whole value is passed to `Integer`.

I can add a test though.

Reply:
added the test